packages/ui/src/views/repo/repo-tags/components/repo-tags-list.tsx

```diff
@@ -1,76 +1,153 @@
-import { CommitCopyActions, MoreActionsTooltip, SkeletonTable, Table } from '@/components'
-import { formatDate } from '@utils/utils'
-import { TranslationStore } from '@views/repo/repo-list/types'
+import { FC, useMemo } from 'react'
 
-import { RepoTagsStore } from '../types'
+import { Avatar, CommitCopyActions, MoreActionsTooltip, NoData, SkeletonTable, Table, Text } from '@/components'
+import { getInitials, timeAgo } from '@/utils'
+import { BranchSelectorListItem, CommitTagType, RepoTagsStore, TranslationStore } from '@/views'
 
 interface RepoTagsListProps {
   useTranslationStore: () => TranslationStore
-  isLoading: boolean
   onDeleteTag: (tagName: string) => void
   useRepoTagsStore: () => RepoTagsStore
   toCommitDetails?: ({ sha }: { sha: string }) => string
-  openCreateBranchDialog: () => void
+  openCreateBranchDialog: (selectedTagInList: BranchSelectorListItem) => void
+  isLoading: boolean
+  isDirtyList: boolean
```

Comment:
As a general rule of thumb, boolean props should be optional and defaulted to `false` to allow for their existence to be `true` and omission to be `false`
```suggestion
  isLoading?: boolean
  isDirtyList?: boolean
```